src/backend/mpi/swdsm.cpp

```diff
@@ -312,19 +314,18 @@ void handler(int sig, siginfo_t *si, void *unused){
 
 	unsigned long tag;
 	argo_byte owner,state;
-	unsigned long distrAddr =  (unsigned long)((unsigned long)(si->si_addr) - (unsigned long)(startAddr));
-
-	unsigned long alignedDistrAddr = alignAddr(distrAddr);
-	unsigned long remCACHELINE = alignedDistrAddr % (CACHELINE*pagesize);
-	unsigned long lineAddr = alignedDistrAddr - remCACHELINE;
-	unsigned long classidx = get_classification_index(lineAddr);
-
-	unsigned long * localAlignedAddr = (unsigned long *)((char*)startAddr + lineAddr);
-	unsigned long startIndex = getCacheIndex(lineAddr);
-	alignedDistrAddr = lineAddr;
-
-	unsigned long homenode = getHomenode(lineAddr);
-	unsigned long offset = getOffset(lineAddr);
+	/* compute offset in distributed memory in bytes, always positive */
+	std::size_t access_offset = static_cast<char*>(si->si_addr) - static_cast<char*>(startAddr);
+
+	/* align access offset to cacheline */
+	std::size_t aligned_access_offset = align_backwards(access_offset, CACHELINE*pagesize);
+	unsigned long classidx = get_classification_index(aligned_access_offset);
+
+	/* compute start pointer of cacheline. char* has byte-wise arithmetics */
+	char* aligned_access_ptr = reinterpret_cast<char*>(startAddr) + aligned_access_offset;
```

Comment:
Why is none of you complaining about this `reinterpret_cast` that should be a `static_cast`?

Reply:
I didn't notice :)

Reply:
fixed